### PR TITLE
Set replace to false when sending Vary header

### LINKED ISSUE #1293: Don't Overwrite Existing Vary Headers
## Expected/Desired Behavior

In a few spots, we send a `Vary` header, but without passing `false` as the 2nd param to `header()`. This means that if there was already a `Vary` header set, it could be overwritten and the request may not be properly vary-ed.

Sending `false` allows multiple `Vary` headers to be sent, which upstream caches can use to properly vary the request.

## Actual Behavior

We overwrite existing `Vary` headers with our `header()` calls.

### PR DESCRIPTION
**Pull request by mjangda**
This is to avoid over-writing `Vary` headers that may have already been sent by other bits of code.

Fixes #1293 

## Checklist

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- n/a This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Add the following to client-mu-plugins

```
<?php

if ( isset( $_SERVER['HTTP_ORIGIN'] ) ) {
	header( 'Vary: Origin' );

	// Load the VIP Vary_Cache class
	require_once( WP_CONTENT_DIR . '/mu-plugins/cache/class-vary-cache.php' );

	use Automattic\VIP\Cache\Vary_Cache;

	// Register the `beta` group
	Vary_Cache::register_group( 'beta' );

	var_dump( $_SERVER['HTTP_ORIGIN'] );

	var_dump( $_COOKIE['vip-go-seg'] );

	add_action( 'template_redirect', function() {
		die( 'all done' );
	}, PHP_INT_MAX );
}
```

Run the following:

```
curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.ca' -H 'YOUR SANDBOX HEADER HERE'

# => miss + example.ca + yep

curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.ca' -H 'YOUR SANDBOX HEADER HERE'

# => hit + example.ca + yep

# change Origin to example.co
curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.co' -H 'YOUR SANDBOX HEADER HERE'

# => hit  + example.ca + yep :( :( :( <= this should have been varied and a miss
```

Now, apply patch:

```
curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.ca' -H 'YOUR SANDBOX HEADER HERE'

# => miss + example.ca + yep

curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.ca' -H 'YOUR SANDBOX HEADER HERE'

# => hit + example.ca + yep

# change Origin to example.co
curl 'https://<SITE_URL>/?bust=1' -H 'Cookie: vip-go-seg=yep' -sv -H 'Origin: http://example.co' -H 'YOUR SANDBOX HEADER HERE'

# => miss + example.co + yep :) :) :)
```

Changing Cookie value and Origin should correct vary cache now. Not that `Vary` header is not sent through but instead `cache-control: private, no-store` so Vary is not necessary.

### CODE REVIEW

File: cache/class-vary-cache.php [hunk 1]
```diff
@@ -585,9 +585,9 @@ private static function send_vary_headers() {
...
 
 			if ( self::is_encryption_enabled() ) {
-				header( 'Vary: X-VIP-Go-Auth' );
+				header( 'Vary: X-VIP-Go-Auth', false );
 			} else {
-				header( 'Vary: X-VIP-Go-Segmentation' );
+				header( 'Vary: X-VIP-Go-Segmentation', false );
 			}
 
 			if ( defined( 'WP_DEBUG' ) && true === WP_DEBUG ) {
```
Comment (dchymko) [inline cache/class-vary-cache.php:593]: Probably want to add the `false` to the header() call here too.

Comment (mjangda) [reply cache/class-vary-cache.php:593]: Not strictly necessary (since it's unlikely that someone else would send the debug header), but wouldn't hurt.

**Comment by mjangda** [comment]
Testing instructions need to be re-done.

**Comment by mjangda** [comment]
Updated testing instructions and verified everything working as expected; this is good to go.

**Comment by mjangda** [comment]
r1179-stacks

**Pull request merged**